api/lib/domain/read-models/CertificationDetails.js

```diff
@@ -20,8 +23,72 @@ class CertificationDetails {
     this.competencesWithMark = competencesWithMark;
     this.listChallengesAndAnswers = listChallengesAndAnswers;
   }
+
+  static from({
+    certificationAssessment,
+    competenceMarks,
+    placementProfile,
+  }) {
+    const reproducibilityRate = ReproducibilityRate.from({ answers: certificationAssessment.certificationAnswersByDate });
+    const competencesWithMark = _buildCompetencesWithMark({ competenceMarks, placementProfile });
+    const listChallengesAndAnswers = _buildListChallengesAndAnswers({ certificationAssessment, competencesWithMark });
+
+    return new CertificationDetails({
+      id: certificationAssessment.certificationCourseId,
+      userId: certificationAssessment.userId,
+      createdAt: certificationAssessment.createdAt,
+      completedAt: certificationAssessment.completedAt,
+      status: certificationAssessment.state,
+      totalScore: _.sumBy(competenceMarks, 'score'),
+      percentageCorrectAnswers: reproducibilityRate.value,
+      competencesWithMark,
+      listChallengesAndAnswers,
+    });
+  }
+}
+
+function _buildCompetencesWithMark({
+  competenceMarks,
+  placementProfile,
+}) {
+  return _.map(competenceMarks, (competenceMark)=> {
+    const userCompetence = placementProfile.getUserCompetence(competenceMark.competenceId);
+
+    return {
+      areaCode: competenceMark.area_code,
+      id: competenceMark.competenceId,
+      index: competenceMark.competence_code,
+      name: userCompetence.name,
+      obtainedLevel: competenceMark.level,
+      obtainedScore: competenceMark.score,
+      positionedLevel: userCompetence.estimatedLevel,
+      positionedScore: userCompetence.pixScore,
+    };
+  });
+
+}
+
+function _buildListChallengesAndAnswers({
+  certificationAssessment,
+  competencesWithMark,
+}) {
+  return _(certificationAssessment.certificationChallenges)
+    .map((certificationChallenge) => {
+      const answerForChallenge = certificationAssessment.findCertificationAnswerForChallenge(certificationChallenge.challengeId);
+      if (!answerForChallenge) return null;
+      const competenceWithMark = _.find(competencesWithMark, { id: certificationChallenge.competenceId });
+      const competenceIndex = competenceWithMark ? competenceWithMark.index : '';
```

Comment:
_getCompetenceIndexForChallenge(certificationChallenge, competenceWithMarks) {
	const competenceWithMark = _.find(competencesWithMark, { id: certificationChallenge.competenceId });
    const competenceIndex = competenceWithMark ? competenceWithMark.index : '';
}

Reply:
`const competenceIndex = this._getCompetenceIndexForChallenge(certificationChallenge, competenceWithMarks);`